publishing-artifacts.md

```diff
@@ -7,33 +7,36 @@ You may obtain a copy of the License at
+Pravega/Flink connector artifacts are used by projects like [pravega-samples](https://github.com/pravega/pravega-samples) which are used by external users. The `master` branch of the [pravega-samples](https://github.com/pravega/pravega-samples) repository will always point to a stable release version of Pravega/Flink connector.
 
-A typical version label of a snapshot artifact will have the label `-SNAPSHOT` associated with it (for e.g., `0.1.0-SNAPSHOT`). There could be more revisions for a snapshot version and the maven/gradle build scripts takes care of fetching most recent version from the available list of revisions. The snapshot repositories are usually configured to discard the old revisions based on some retention policy settings. Any downstream projects that are depending on snapshots are faced with the challenge of keeping up with the snapshot changes since they could possibly introduce any breaking changes.
+ However, the development branch (`develop`) of [pravega-samples](https://github.com/pravega/pravega-samples/tree/develop) are likely to be unstable due to the Pravega/Flink connector snapshot artifact dependency. (_As the development branch of Pravega/Flink connector could possibly introduce a breaking change_).
 
-To overcome this issue, the Pravega/Flink connector snapshot artifacts are published to `jcenter` repository with a **stable** version label which can be referred by any downstream projects. This will guarantee to avoid any breaking changes from Pravega/Flink connector but at the same time, it is the responsibility of the downstream projects to sync with the latest shapshot updates.
+A typical version label of a snapshot artifact will have the label `-SNAPSHOT` associated with it (for e.g., `0.1.0-SNAPSHOT`). There could be more revisions for a snapshot version and the `maven/gradle` build scripts is responsible for fetching the most recent version from the available list of revisions. The snapshot repositories are usually configured to discard the old revisions based on some retention policy settings. Any downstream projects that are depending on snapshots are faced with the challenges of keeping up with the snapshot changes since they could possibly introduce any breaking changes.
 
+To overcome this issue, the Pravega/Flink connector snapshot artifacts are published to `jcenter` repository with a **stable** version label which can be referred by any downstream projects. This will guarantee to avoid any breaking changes from Pravega/Flink connector, but at the same time, it is the responsibility of the downstream projects to sync with the latest snapshot updates.

```

Comment:
We can rephrase that to the following.

> To overcome this issue, the Pravega/Flink connector snapshot artifacts are published to `jcenter` repository with a **stable** version label which can be referred by any downstream projects. This will guarantee the build stability of the downstream projects that has a dependency on `flink-connectors` snapshot artifacts. However, unlike typical maven snapshots that are refreshed automatically upon any new revisions, the downstream projects are expected to synchronize with any latest snapshot revisions from `flink-connectors` by manually updating the build to make use of the new revisions.

Reply:
good thank you!